src/ghostfat.c

```diff
@@ -441,38 +416,35 @@ void uf2_read_block (uint32_t block_no, uint8_t *data)
           fileIndex++, d++ )
     {
       // WARNING -- code presumes all files take exactly one directory entry (no long file names!)
-      uint32_t const startCluster = info_cluster_start(fileIndex);
+      uint32_t const startCluster = info[fileIndex].cluster_start;
 
       FileContent_t const *inf = &info[fileIndex];
       padded_memcpy(d->name, inf->name, 11);
-      d->createTimeFine = COMPILE_SECONDS_INT % 2 * 100;
-      d->createTime = COMPILE_DOS_TIME;
-      d->createDate = COMPILE_DOS_DATE;
-      d->lastAccessDate = COMPILE_DOS_DATE;
+      d->createTimeFine   = COMPILE_SECONDS_INT % 2 * 100;
+      d->createTime       = COMPILE_DOS_TIME;
+      d->createDate       = COMPILE_DOS_DATE;
+      d->lastAccessDate   = COMPILE_DOS_DATE;
       d->highStartCluster = startCluster >> 16;
-      d->updateTime = COMPILE_DOS_TIME;
-      d->updateDate = COMPILE_DOS_DATE;
-      d->startCluster = startCluster & 0xFFFF;
-      d->size = (inf->content ? inf->size : UF2_BYTE_COUNT);
+      d->updateTime       = COMPILE_DOS_TIME;
+      d->updateDate       = COMPILE_DOS_DATE;
+      d->startCluster     = startCluster & 0xFFFF;
+      d->size             = (inf->content ? inf->size : UF2_BYTE_COUNT);
     }
   }
-  else if ( block_no < BPB_TOTAL_SECTORS ) // aka read from the data area (files, unused space, ...)
+  else if ( block_no < BPB_TOTAL_SECTORS )
   {
-
+    // aka read from the data area (files, unused space, ...)
     sectionRelativeSector -= FS_START_CLUSTERS_SECTOR;
 
     // plus 2 for first data cluster offset
-    uint32_t const cluster_num = 2 + sectionRelativeSector / BPB_SECTORS_PER_CLUSTER;
+    uint32_t fid = info_index_of(2 + sectionRelativeSector / BPB_SECTORS_PER_CLUSTER);
+    FileContent_t const * inf = &info[fid];
 
-    if ( cluster_num < UF2_FIRST_CLUSTER_NUMBER )
+    sectionRelativeSector -= (info[fid].cluster_start-2) * BPB_SECTORS_PER_CLUSTER;
```

Comment:
Don't re-use `sectionRelativeSector` ... create new variable, such as `fileRelativeSector`.  The compiler will optimize it anyways, but it will be much clearer what value is stored....

Reply:
thank you